packages/analytics-browser/src/browser-client.ts

```diff
@@ -44,7 +44,7 @@ export class AmplitudeBrowser extends AmplitudeCore<BrowserConfig> {
     return this.config.userId;
   }
 
-  setUserId(userId: string) {
```

Comment:
nope, but making it `undefined` should effectively do the same thing. i chose `undefined` instead of null since using `undefined` is more intuitive in typescript

Reply:
But we previously accept setUserId(null), do we still need to support that?

Reply:
Hmm I don't see any added benefits for supporting both. With type definition, `undefined` should be enforced by the TS transpiler to assist developers. If they are on a JS package, `null` should work also. Also to be clear, `ts-browser` never accepted null as user id.